Disposable email project.

pradzikowski opened the pull request.

pradzikowski committed:
Init.
Whole code structure without tests.

pradzikowski commented:
![](http://www.catgifpage.com/gifs/318.gif)

pradzikowski commented:
@adeptofvoltron i was end my WIP process. I will wait for your review

adeptofvoltron commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+  - composer install --prefer-dist --no-interaction
+
+script:
+  - phpunit
```
more likely to use `./vendor/bin/phpunit` than you will make sure the travis will run proper version of phpunit

pradzikowski replied:
Sure, I will fix this.

adeptofvoltron commented on src/Exception/DisposableEmailException.php:
```diff
@@ -0,0 +1,8 @@
+namespace PCF\DisposableEmail\Exception;
+
+class DisposableEmailException extends \Exception
+{}
```
Opening class brace must be on a line by itself
Closing class brace must be on a line by itself

adeptofvoltron replied:
similar on VerifyDomainException

pradzikowski replied:
👌

adeptofvoltron commented on src/Resource/BlockList.php:
```diff
@@ -0,0 +1,2187 @@
+zzz.com
+LIST;
+
+}
```
line after closing brace

adeptofvoltron commented on src/Resource/BlockList.php:
```diff
@@ -0,0 +1,2187 @@
+zxcvbnm.com
+zzz.com
+LIST;
+
```
useless empty line

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+     */
+    public function verifyDomain(string $domain): int
+    {
+        if(empty($this->collection)) {
```
space after if keyword

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+            throw new VerifyDomainException('DomainCollection does not set');
+        }
+
+        if(empty($domain)) {
```
space after if keyword

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+     */
+    private function pregList(array $list, string $domain): bool
+    {
+        if(empty($list)) {
```
space after if keyword

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+        ];
+
+        foreach ($lists as $status => $list) {
+            if(true === $this->pregList($list, $domain)) {
```
space after if keyword

pradzikowski replied:
done

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+        }
+
+        foreach ($list as $pattern) {
+            if(true == preg_match($pattern, $domain)) { //IHATETHISFUNCTIONBECAUSEITRETURNINTINTEADBOOLEAN
```
space after if keyword

adeptofvoltron commented on src/EmailVerifierInterface.php:
```diff
@@ -0,0 +1,34 @@
+ *
+ * @package PCF\DisposableEmail\Verifier
+ */
+Interface EmailVerifierInterface
```
`I` => `i`

adeptofvoltron commented on src/ListedVerifier.php:
```diff
@@ -0,0 +1,80 @@
+
+declare(strict_types=1);
+
+namespace PCF\DisposableEmail;
```
line after namespace

pradzikowski replied:
done

adeptofvoltron commented on src/ListedVerifier.php:
```diff
@@ -0,0 +1,80 @@
+     */
+    public function verifyDomain(string $domain): int
+    {
+        if(empty($this->collection)) {
```
space after if keyword

adeptofvoltron commented on src/ListedVerifier.php:
```diff
@@ -0,0 +1,80 @@
+            throw new VerifyDomainException('DomainCollection does not set');
+        }
+
+        if(empty($domain)) {
```
space after if keyword

adeptofvoltron commented on src/ListedVerifier.php:
```diff
@@ -0,0 +1,80 @@
+        ];
+
+        foreach ($lists as $status => $list) {
+            if(empty($list)) {
```
space after if keyword

adeptofvoltron commented on src/ListedVerifier.php:
```diff
@@ -0,0 +1,80 @@
+                continue;
+            }
+
+            if(in_array($domain, $list)){
```
space after if keyword and one before brace

adeptofvoltron commented on test/RegexVerifierTest.php:
```diff
@@ -0,0 +1,77 @@
+declare(strict_types=1);
+
+namespace PCF\DisposableEmailTest;
+
```
expected one blank line...not two

pradzikowski replied:
done

adeptofvoltron commented on test/RegexVerifierTest.php:
```diff
@@ -0,0 +1,77 @@
+            [ $collection, 'domain7.test', RegexVerifier::DOMAIN_UNKNOWN   ],
+        ];
+    }
+}
```
expected one blank line at the end of file

pradzikowski replied:
done

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+        }
+
+        foreach ($list as $pattern) {
+            if(true == preg_match($pattern, $domain)) { //IHATETHISFUNCTIONBECAUSEITRETURNINTINTEADBOOLEAN
```
`HP Warning:  preg_match(): Delimiter must not be alphanumeric or backslash in /home/praca/Pulpit/DisposableEmail/src/RegexVerifier.php on line 83`

pradzikowski replied:
Could you show me you runtime code ? what pattern you use ?

adeptofvoltron commented on src/RegexVerifier.php:
```diff
@@ -0,0 +1,90 @@
+        }
+
+        foreach ($list as $pattern) {
+            if(true == preg_match($pattern, $domain)) { //IHATETHISFUNCTIONBECAUSEITRETURNINTINTEADBOOLEAN
```
useless comment

pradzikowski replied:
stfu :D

adeptofvoltron reviewed (changes requested):
till `verifyDomain` is not static method, it would be nice(but not must have) to create Factry for Listed/RegexVerifier.

other hand you can create some static methods inside both methods...that allow you to verifyDomain. All dependiences are inside package.

adeptofvoltron commented on src/Resource/AbstractResourceList.php:
```diff
@@ -0,0 +1,25 @@
+     */
+    public static function getList(): array
+    {
+        return explode(PHP_EOL, trim(static::$list));
```
topic to discuss: You can copy list from [source](https://github.com/martenson/disposable-email-domains) as a text. But in other hands, thi lib will be more efficient if we would allready keep list as array

pradzikowski replied:
I think about it and yes i will clone source files as original. Imo this is not reasonable to be afraid about perfomance and efficient in this case. Users and frameworks can cache this data in faster storages than static txt. 

Current frameworks have fine mechanism for caching static resources (ex. routing)for example [syfony](http://symfony.com/doc/current/components/dependency_injection/compilation.html)

pradzikowski commented:
@adeptofvoltron Imo i dont need to create factories for these classes because their implementation deeply based on users code. Of course i can create factory only for our resources but i dont think we need this. Maybe i will create builder, but i dont promise.

pradzikowski commented:
https://www.catgifpage.com/cute

@adeptofvoltron done :)

pradzikowski committed:
Added block and trust list with some PHP code implementations.
Placed block and trust list in src/Resource/lists
Placed AbstractResourceList in src/Resource/lists for easy files reading.
Created interface DomainCollectionInterface which can contains list of trusted, blocked and knowed domains, regex of these domain, etc.
Created SimpleDomainCollection which implement DomainCollectionInterface.
Created EmailVerifierInterface which contains visualisation of Verifier conspect.
Created AbstractPCFVerifier which implement EmailVerifierInterface. It contains all what user needs for proper test disposable email domains.
Created ListedVerifier which extends AbstractPCFVerifier. Its easy class which check is current checked domain in lists.
Created RegexVerifier which extends AbstractPCFVerivier. It check is current checked domain is in one of regular expression.
